app/javascript/tests/public_health/bulk_actions/CloseRecords.test.js

```diff
@@ -0,0 +1,97 @@
+const mockToken = 'testMockTokenString12345';
+const patientsArray = [mockPatient1, mockPatient2, mockPatient6];
+
+function getWrapper() {
+  return shallow(<CloseRecords authenticity_token={mockToken} patients={patientsArray} close={onCloseMock} monitoring_reasons={mockMonitoringReasons} />);
+}
+
+describe('CloseRecords', () => {
+  it('Properly renders all main components', () => {
+    const wrapper = getWrapper();
+    expect(wrapper.find('p').find('span').at(0).text()).toEqual('You are about to change the Monitoring Status of the selected records from "Actively Monitoring" to "Not Monitoring".');
+    expect(wrapper.find('p').find('span').at(1).text()).toEqual(' These records will be moved to the closed line list and the reason for closure will be blank.');
+    expect(wrapper.find(Form.Group).length).toEqual(3);
+    expect(wrapper.find(Form.Group).at(0).find(Form.Label).text()).toEqual('Please select reason for status change:');
+    expect(wrapper.find('#monitoring_reason').exists()).toBeTruthy();
+    const monitoringReasonOptions = [''].concat(mockMonitoringReasons);
+    wrapper
+      .find('#monitoring_reason')
+      .find('option')
+      .forEach((option, index) => {
+        expect(option.text()).toEqual(monitoringReasonOptions[Number(index)]);
+      });
+    expect(wrapper.find(Form.Group).at(1).find(Form.Label).length).toEqual(2);
+    expect(wrapper.find(Form.Group).at(1).find(Form.Label).at(0).text()).toEqual('Please include any additional details:');
+    expect(wrapper.find(Form.Group).at(1).find(Form.Label).at(1).text()).toContain('characters remaining');
+    expect(wrapper.find('#apply_to_household').exists()).toBeTruthy();
+    expect(wrapper.find(Modal.Footer).find(Button).at(0).text()).toEqual('Cancel');
+    expect(wrapper.find(Modal.Footer).find(Button).at(1).text()).toEqual('Submit');
+  });
+
+  it('Changing monitoring reason dropdown updates state', () => {
+    const wrapper = getWrapper();
+    // initial modal state with monitoring reason empty
+    expect(wrapper.state('monitoring_reason')).toEqual('');
+    expect(wrapper.find('p').text()).toContain(' These records will be moved to the closed line list and the reason for closure will be blank.');
+
+    // test changing to each enabled monitoring option
+    mockMonitoringReasons.forEach(value => {
+      wrapper
+        .find(Form.Control)
+        .at(0)
+        .simulate('change', { target: { id: 'monitoring_reason', value: value } });
+      expect(wrapper.state('monitoring_reason')).toEqual(value);
+      expect(wrapper.state('monitoring')).toEqual(false);
+      expect(wrapper.find('p').text()).not.toContain(' These records will be moved to the closed line list and the reason for closure will be blank.');
+    });
+  });
+
+  it('Correctly updates the character limit label for the reason', () => {
+    const wrapper = getWrapper();
+    expect(wrapper.find('.character-limit-text').first().text()).toContain('2000 characters remaining');
```

Comment:
this can be a `toEqual`

Reply:
and you shouldn't need the `.first()` here either

Reply:
Completely disagree @sgober  `toContain()` is absolutely the correct choice here, for the reason shown by @hackrm.

Reply:
So the reasoning behind this is because there are extra spaces in this div [here](https://github.com/SaraAlert/SaraAlert/blob/1537-bulk-action-tests/app/javascript/components/public_health/bulk_actions/CloseRecords.js#L94) which are totally unnecessary.  I would remove the spaces in the html and use `toEqual` here